src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -40,19 +50,55 @@
 public class DomainApis {
 
     private static final Logger logger = Logger.getLogger(DomainApis.class);
+    private EntityManager em;
+    private TransactionControl txControl;
+
+    /**
+     * init the DB
+     *
+     *
+     */
+    public void init(EntityManager em, TransactionControl txControl) throws Exception {
+        this.em = em;
+        this.txControl = txControl;
+    }
 
     /**
      * Creates the Policy for a given domain
      *
      * @return policy
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @POST
     public PolicyEntity createPolicy(@PathParam("domainid") Long domainid, PolicyEntity entity) {
 
-        logger.info("Creating Policy Entity...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Creating Policy Entity...:" + entity.getName());
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<DomainEntity> q = cb.createQuery(DomainEntity.class);
+                Root<DomainEntity> r = q.from(DomainEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("Id"), domainid)));
+
+                List<DomainEntity> result = em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    PolicyEntity policy = new PolicyEntity();
```

Comment:
What error does the REST api return when a policy is being added with the same name of one that already might exist? You are enforcing the name is unique so i am guessing this case will cause 500, ensure that if a policy with the same name already exists you return bad request rather than 500. Also log the reason for the failure

Reply:
OK